client/packages/coldchain/src/ColdchainNotification.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { PropsWithChildren } from 'react';
+import {
+  BaseButton,
+  Box,
+  RouteBuilder,
+  TemperatureBreachSortFieldInput,
+  Typography,
+  useMatch,
+  useNavigate,
+  useUrlQuery,
+} from '@openmsupply-client/common';
+import { useFormatDateTime, useTranslation } from '@common/intl';
+import { CircleAlertIcon } from '@common/icons';
+import { alpha, useTheme } from '@common/styles';
+import { AppRoute } from '@openmsupply-client/config';
+import {
+  TemperatureBreachFragment,
+  useTemperatureBreach,
+} from './Monitoring/api/TemperatureBreach';
+
+const Text: React.FC<PropsWithChildren> = ({ children }) => (
+  <Typography
+    component="div"
+    sx={{
+      fontSize: '14px',
+      display: 'flex',
+      alignContent: 'center',
+      flexWrap: 'wrap',
+    }}
+  >
+    {children}
+  </Typography>
+);
+
+const Separator = () => (
+  <Text>
+    <Typography paddingX={0.5}>|</Typography>
+  </Text>
+);
+
+const DetailButton = ({ breach }: { breach: TemperatureBreachFragment }) => {
+  const t = useTranslation('coldchain');
+  const navigate = useNavigate();
+  const { urlQuery } = useUrlQuery();
+  const currentTab = (urlQuery['tab'] as string) ?? '';
+  const isColdchain = useMatch(
+    RouteBuilder.create(AppRoute.Coldchain).addWildCard().build()
+  );
+
+  if (isColdchain && currentTab === t('label.breaches')) return null;
+
+  return (
+    <BaseButton
+      variant="contained"
+      style={{ height: 32 }}
+      onClick={() =>
+        navigate(
+          RouteBuilder.create(AppRoute.Coldchain)
+            .addPart(AppRoute.Monitoring)
+            .addQuery({ tab: t('label.breaches') })
+            .addQuery({
+              sort: TemperatureBreachSortFieldInput.StartDatetime,
+            })
+            .addQuery({ acknowledged: false })
+            .addQuery({ sensorId: breach?.sensor?.id ?? '' })
+            .build()
+        )
+      }
+    >
+      {t('button.view-details')}
+    </BaseButton>
+  );
+};
+const Location = ({ breach }: { breach: TemperatureBreachFragment }) => {
+  const t = useTranslation('coldchain');
+
+  if (!breach?.location?.name) return null;
+  return (
+    <>
+      <Separator />
+      {!!breach?.location?.name && (
+        <Text>
+          {t('message.location')}
+          <b style={{ paddingLeft: 4 }}>{breach?.location?.name}</b>
+        </Text>
```

Comment:
I think you can remove all the `undefined` breach.location.name checks since you return early if the name is not set.

Reply:
thanks, tidied up